python/delta_sharing/rest_client.py

```diff
@@ -274,3 +279,14 @@ def _should_retry(self, error):
             return True
         else:
             return False
+
+    def _error_on_expired_token(self, error):
+        token_expired = False
```

Comment:
Maybe write it in this way to avoid parsing the expiration time when it's not 401
```
if isinstance(error, HTTPError) and error.response.status_code == 401:
  try:
            expiration_time = datetime.fromisoformat(self._profile.expiration_time)
            return datetime.now() > expiration_time
        except Exception:
            return False
else:
  return False
```
- `fromisotime` should be `fromisoformat`
- We should use `401 Unauthorized` rather than `403 Forbidden` .